aviatrix/resource_aviatrix_controller_cert_domain_config.go

```diff
@@ -37,7 +37,18 @@ func resourceAviatrixControllerCertDomainConfigCreate(ctx context.Context, d *sc
 
 	err := client.SetCertDomain(ctx, certDomain)
 	if err != nil {
-		return diag.Errorf("could not set cert domain: %v", err)
+		if strings.Contains(err.Error(), "EOF") {
+			certDomainConfig, err := client.GetCertDomain(ctx)
+			if err != nil {
+				return diag.Errorf("could not get cert domain: %v", err)
+			}
+			if certDomainConfig.CertDomain == certDomain {
+				d.SetId(strings.Replace(client.ControllerIP, ".", "-", -1))
+				return resourceAviatrixControllerCertDomainConfigRead(ctx, d, meta)
```

Comment:
Also we should not return early if the operation is successful, just let the function return at the bottom of the function